### PR TITLE
Remove sym from gsMatrixOp; add unittest

### PR DESCRIPTION
**Pull request by stefantakacs**
This now only consists of removing sym and unittests

### CODE REVIEW

File: src/gsSolver/gsMatrixOp.h [hunk 1]
```diff
@@ -49,42 +49,40 @@ class gsMatrixOp : public gsLinearOperator<typename MatrixType::Scalar>
...
+    : m_mat(), m_expr(mat.derived())
     {
         //gsDebug<<typeid(m_expr).name()<<" Ref: "<<is_ref<NestedMatrix>::value<<"\n";
     }
 
     /// @brief Constructor taking a shared pointer
-    gsMatrixOp(const MatrixPtr& mat, bool sym=false)
-    : m_mat(mat), m_expr(m_mat->derived()), m_symmetric(sym)
+    gsMatrixOp(const MatrixPtr& mat)
+    : m_mat(mat), m_expr(m_mat->derived())
```
Comment (filiatra) [inline src/gsSolver/gsMatrixOp.h]: I think that
gsMatrixOp(MatrixPtr mat) : m_mat(give(mat))

is more suitable here

File: src/gsSolver/gsMatrixOp.h [hunk 2]
```diff
@@ -49,42 +49,40 @@ class gsMatrixOp : public gsLinearOperator<typename MatrixType::Scalar>
...
     /// is not deleted too early or provide a shared pointer.
-    static uPtr make(const MatrixType& mat, bool sym=false)
-    { return memory::make_unique( new gsMatrixOp(mat,sym) ); }
+    static uPtr make(const MatrixType& mat)
+    { return memory::make_unique( new gsMatrixOp(mat) ); }
 
     /// Make function returning a smart pointer
-    static uPtr make(const MatrixPtr& mat, bool sym=false)
-    { return memory::make_unique( new gsMatrixOp(mat,sym) ); }
+    static uPtr make(const MatrixPtr& mat)
```
Comment (filiatra) [inline src/gsSolver/gsMatrixOp.h]: similarly, make(MatrixPtr mat)

File: src/gsSolver/gsMatrixOp.h [hunk 3]
```diff
@@ -153,16 +150,16 @@ typename gsMatrixOp<Derived>::uPtr makeMatrixOp(const Eigen::EigenBase<Derived>&
...
+typename gsMatrixOp<Derived>::uPtr makeMatrixOp(const memory::shared_ptr<Derived> & mat)
 {
-    return memory::make_unique(new gsMatrixOp<Derived>(mat, sym));
+    return memory::make_unique(new gsMatrixOp<Derived>(mat));
 }
 
 // We need an additional guide for the compiler to be able to work well with unique ptrs
 template <class Derived>
-typename gsMatrixOp<Derived>::uPtr makeMatrixOp(memory::unique_ptr<Derived> mat, bool sym=false)
+typename gsMatrixOp<Derived>::uPtr makeMatrixOp(memory::unique_ptr<Derived> mat)
```
Comment (filiatra) [inline src/gsSolver/gsMatrixOp.h]: This one should be possible to be removed, when the make(shared_ptr) is updated

**Comment by stefantakacs** [comment]
We cannot merge this PR request as long as the issue with the unittest (cf. PR#118) is not fixed because this is needed in devel.

File: src/gsSolver/gsMatrixOp.h (hunk 3 above)
Comment (stefantakacs) [reply src/gsSolver/gsMatrixOp.h]: no, removing this does not work. here, the problem is not the move semantics but the missing ability of c++ to deduce "Derived".

File: src/gsSolver/gsMatrixOp.h (hunk 2 above)
Comment (stefantakacs) [reply src/gsSolver/gsMatrixOp.h]: done

File: src/gsSolver/gsMatrixOp.h (hunk 1 above)
Comment (stefantakacs) [reply src/gsSolver/gsMatrixOp.h]: done

**Comment by stefantakacs** [comment]
found out that devel can now live without sym and without plugging selfadjoinedView into gsMatrixOp.

**Pull request merged**